app/Http/Controllers/V2/Polygons/ChangeStatusPolygonsController.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App\Http\Controllers\V2\Polygons;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\V2\Polygons\ChangeStatusPolygonsUpdateRequest;
+use App\Models\Traits\SaveAuditStatusTrait;
+use App\Models\V2\Sites\SitePolygon;
+
+class ChangeStatusPolygonsController extends Controller
+{
+    use SaveAuditStatusTrait;
+
+    public function __invoke(ChangeStatusPolygonsUpdateRequest $request)
+    {
+        $body = $request->all();
+        $polygons = $body['updatePolygons'];
+
+        foreach ($polygons as $polygon) {
```

Comment:
Any time I see DB queries being executed in a loop, I want that N+1 query state to be improved. In this case, I think you should be able to do something more like
```
$sitePolygons = SitePolyon::whereIn('uuid', collect($polygons)->map(fn ($p) => $p['uuid'])->get();
foreach ($sitePolygons as $sitePolygon) { 
  ....
```

You might have to do a little extra to find the associated polygon status from the $polygons array, but that's better than issuing a DB query per polygon

Reply:
changed